Auto update failing for qt6-qttools

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [qt6-qttools](https://github.com/termux/termux-packages/blob/bfa829025f2f3f5b0431f64aaf21e03c2954e900/x11-packages/qt6-qttools) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating qt6-qttools [Current version: 6.8.3]
INFO: package being updated to 6.9.0.
Downloading https://download.qt.io/official_releases/qt/6.9/6.9.0/submodules/qttools-everywhere-src-6.9.0.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   282  100   282    0     0    500      0 --:--:-- --:--:-- --:--:--   500

100  9.8M  100  9.8M    0     0  13.3M      0 --:--:-- --:--:-- --:--:-- 13.3M

You are about to commit these changes:

--------------------
diff --git a/x11-packages/qt6-qttools/build.sh b/x11-packages/qt6-qttools/build.sh
index 24b25bdd3c..f1217f05ce 100644
--- a/x11-packages/qt6-qttools/build.sh
+++ b/x11-packages/qt6-qttools/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE=https://www.qt.io/
 TERMUX_PKG_DESCRIPTION="Qt Development Tools (Linguist, Assistant, Designer, etc.)"
 TERMUX_PKG_LICENSE="GPL-3.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="6.8.3"
+TERMUX_PKG_VERSION="6.9.0"
 TERMUX_PKG_SRCURL="https://download.qt.io/official_releases/qt/${TERMUX_PKG_VERSION%.*}/${TERMUX_PKG_VERSION}/submodules/qttools-everywhere-src-${TERMUX_PKG_VERSION}.tar.xz"
-TERMUX_PKG_SHA256=02a4e219248b94f1333df843d25763f35251c1074cdc4fb5bda67d340f8c8b3a
+TERMUX_PKG_SHA256=fa645589cc3f939022401a926825972a44277dead8ec8607d9f2662e6529c9a4
 TERMUX_PKG_DEPENDS="libc++, qt6-qtbase (>= ${TERMUX_PKG_VERSION}), zstd"
 TERMUX_PKG_BUILD_DEPENDS="qt6-qtdeclarative (>= ${TERMUX_PKG_VERSION})"
 TERMUX_PKG_RECOMMENDS="qt6-qtdeclarative"
--------------------

bump(x11/qt6-qttools): 6.9.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package qt6-qttools exists in x11 repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
Creating new container...
docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?.
See 'docker run --help'.
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2025-04-02 18:22:29 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/14226428654">14226428654</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

Can not be updated automatically due to
```
-- The CXX compiler identification is GNU 14.2.0
-- The C compiler identification is GNU 14.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Warning at /data/data/com.termux/files/usr/opt/qt6/cross/lib/cmake/Qt6/Qt6ConfigVersion.cmake:56 (message):
  Version 6.9.0 of package Qt6 was requested but an incompatible version was
  found: 6.8.3.  You can pass -DQT_NO_PACKAGE_VERSION_CHECK=TRUE to disable
  the version check and force the incompatible version to be used.  There is
  no guarantee the build will succeed.  Use at your own risk.  You can
  silence this warning by passing
  -DQT_NO_PACKAGE_VERSION_INCOMPATIBLE_WARNING=TRUE
Call Stack (most recent call first):
  CMakeLists.txt:21 (find_package)


CMake Error at CMakeLists.txt:21 (find_package):
  Could not find a configuration file for package "Qt6" that is compatible
  with requested version "6.9.0".

  The following configuration files were considered but not accepted:

    /data/data/com.termux/files/usr/opt/qt6/cross/lib/cmake/Qt6/Qt6Config.cmake, version: 6.8.3



-- Configuring incomplete, errors occurred!
```
Probably we should update qt6-qtbase package first.

WEel, now (after qt6-qtbase update) the situation is a bit better.
```
CMake Error at /usr/lib/x86_64-linux-gnu/cmake/zstd/zstdTargets.cmake:42 (message):
  Some (but not all) targets in this export set were already defined.

  Targets Defined: zstd::libzstd_shared, zstd::libzstd_static

  Targets not yet defined: zstd::libzstd

Call Stack (most recent call first):
  /usr/lib/x86_64-linux-gnu/cmake/zstd/zstdConfig.cmake:42 (include)
  /data/data/com.termux/files/usr/opt/qt6/cross/lib/cmake/Qt6/FindWrapZSTD.cmake:24 (find_package)
  /data/data/com.termux/files/usr/opt/qt6/cross/lib/cmake/Qt6/QtFindPackageHelpers.cmake:141 (find_package)
  src/designer/src/lib/CMakeLists.txt:425 (qt_find_package)
```

--- a/x11-packages/qt6-qttools/cmake-find-zstd-before-clang.patch.beforehostbuild
+++ b/x11-packages/qt6-qttools/cmake-find-zstd-before-clang.patch.beforehostbuild
@@ -1,0 +1,47 @@
+# https://salsa.debian.org/qt-kde-team/qt6/qt6-tools/-/blob/debian/6.8.2-4/debian/patches/cmake-find-zstd-before-clang.diff
+# Fixes the following error
+# CMake Error at /data/data/com.termux/files/usr/lib/cmake/zstd/zstdTargets.cmake:42 (message):
+# Some (but not all) targets in this export set were already defined.
+# Targets Defined: zstd::libzstd_shared
+# Targets not yet defined: zstd::libzstd
+
+--- a/configure.cmake
++++ b/configure.cmake
+@@ -3,6 +3,19 @@
+
+ #### Tests
+
++# HACK: look for libzstd before Clang/LLVM; this way all the right
++# libzstd cmake targets are defined, instead of the partial ones defined
++# by the Findzstd shipped with LLVM
++if(NOT TARGET WrapZSTD::WrapZSTD)
++    qt_find_package(WrapZSTD 1.3
++        PROVIDED_TARGETS
++            WrapZSTD::WrapZSTD
++            zstd::libzstd
++            zstd::libzstd_static
++            zstd::libzstd_shared
++    )
++endif()
++
+ qt_find_package(WrapLibClang 8 PROVIDED_TARGETS WrapLibClang::WrapLibClang)
+
+ if(TARGET WrapLibClang::WrapLibClang)
+--- a/src/designer/src/lib/CMakeLists.txt
++++ b/src/designer/src/lib/CMakeLists.txt
+@@ -421,15 +421,6 @@
+ if(TARGET zstd::libzstd_static)
+     qt_internal_disable_find_package_global_promotion(zstd::libzstd_static)
+ endif()
+-if(NOT TARGET WrapZSTD::WrapZSTD)
+-    qt_find_package(WrapZSTD 1.3
+-        PROVIDED_TARGETS
+-            WrapZSTD::WrapZSTD
+-            zstd::libzstd
+-            zstd::libzstd_static
+-            zstd::libzstd_shared
+-    )
+-endif()
+
+ qt_internal_extend_target(Designer CONDITION QT_FEATURE_zstd
+     LIBRARIES

